dawn/src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -250,138 +228,124 @@ DawnCompiler::optimize(std::map<std::string, std::shared_ptr<iir::StencilInstant
-    // validation check
-    optimizer.pushBackPass<PassValidation>();
-  }
-  //===-----------------------------------------------------------------------------------------
-  if(shouldRunGroup(options_, options_.SetStageName)) {
-    // This is never used but if we want to reenable it, it is independent
-    optimizer.pushBackPass<PassSetStageName>();
-    // validation check
-    optimizer.pushBackPass<PassValidation>();
-  }
-  //===-----------------------------------------------------------------------------------------
-  if(shouldRunGroup(options_, options_.StageReordering)) {
-    optimizer.pushBackPass<PassSetStageGraph>();
-    optimizer.pushBackPass<PassSetDependencyGraph>();
-    optimizer.pushBackPass<PassStageReordering>(reorderStrategy);
-    // moved stages around ...
-    optimizer.pushBackPass<PassSetSyncStage>();
-    // if we want this info around, we should probably run this also
-    // optimizer.pushBackPass<PassSetStageName>();
-    // validation check
-    optimizer.pushBackPass<PassValidation>();
-  }
-  //===-----------------------------------------------------------------------------------------
-  if(shouldRunGroup(options_, options_.StageMerger)) {
-    // merging requires the stage graph
-    optimizer.pushBackPass<PassSetStageGraph>();
-    // running the actual pass
-    optimizer.pushBackPass<PassStageMerger>();
-    // since this can change the scope of temporaries ...
-    optimizer.pushBackPass<PassTemporaryType>();
-    optimizer.pushBackPass<PassLocalVarType>();
-    optimizer.pushBackPass<PassRemoveScalars>();
-    // modify stages and their extents ...
-    optimizer.pushBackPass<PassComputeStageExtents>();
-    // and changes their dependencies
-    optimizer.pushBackPass<PassSetSyncStage>();
-    // validation check
-    optimizer.pushBackPass<PassValidation>();
-  }
-  //===-----------------------------------------------------------------------------------------
-  // // should be irrelevant now
-  // optimizer.pushBackPass<PassStencilSplitter>(maxFields);
-  // // but would require a lot
-  //===-----------------------------------------------------------------------------------------
-  if(shouldRunGroup(options_, options_.TemporaryMerger)) {
-    optimizer.pushBackPass<PassTemporaryMerger>();
-    // validation check
-    optimizer.pushBackPass<PassValidation>();
-  }
-  //===-----------------------------------------------------------------------------------------
-  if(shouldRunGroup(options_, options_.Inlining)) {
-    optimizer.pushBackPass<PassInlining>(
-        (getOptions().Inlining || getOptions().TmpToStencilFunction),
-        PassInlining::InlineStrategy::ComputationsOnTheFly);
-    // validation check
-    optimizer.pushBackPass<PassValidation>();
-  }
-  //===-----------------------------------------------------------------------------------------
-  if(shouldRunGroup(options_, options_.IntervalPartitioning)) {
-    if(options_.IntervalPartitioning) {
+  if(groups.empty())
```

Comment:
shouldn't this be the responsibility of the caller of optimize? Feels out of place here

Reply:
Will be done in a follow-up PR.